atd-etl/cris_import/cris_import.py

```diff
@@ -0,0 +1,801 @@
+            "opvault": VAULT_ID,
+        },
+        "database_username": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.Database Username",
+            "opvault": VAULT_ID,
+        },
+        "database_password": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.Database Password",
+            "opvault": VAULT_ID,
+        },
+        "database_name": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.Database Name",
+            "opvault": VAULT_ID,
+        },
+        "database_ssl_policy": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.Database SSL Policy",
+            "opvault": VAULT_ID,
+        },
+        "aws_access_key": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.AWS Access key",
+            "opvault": VAULT_ID,
+        },
+        "aws_secret_key": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.AWS Secret key",
+            "opvault": VAULT_ID,
+        },
+        "s3_archive_bucket_name": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.S3 Archive Bucket Name",
+            "opvault": VAULT_ID,
+        },
+        "s3_archive_path": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.S3 Archive Path",
+            "opvault": VAULT_ID,
+        },
+        "graphql_endpoint": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.GraphQL Endpoint",
+            "opvault": VAULT_ID,
+        },
+        "graphql_endpoint_key": {
+            "opitem": "Vision Zero CRIS Import",
+            "opfield": f"{DEPLOYMENT_ENVIRONMENT}.GraphQL Endpoint key",
+            "opvault": VAULT_ID,
+        },
+        "sftp_endpoint_private_key": {
+            "opitem": "SFTP Endpoint Key",
+            "opfield": ".private key",
+            "opvault": VAULT_ID,
+            },
+    }
+
+    # instantiate a 1Password client
+    client: Client = new_client(ONEPASSWORD_CONNECT_HOST, ONEPASSWORD_CONNECT_TOKEN)
+    # get the requested secrets from 1Password
+    return onepasswordconnectsdk.load_dict(client, REQUIRED_SECRETS)
+
+
+def specify_extract_location(file):
+    zip_tmpdir = tempfile.mkdtemp()
+    shutil.copy(file, zip_tmpdir)
+    return zip_tmpdir
+
+
+def download_extract_archives():
+    """
+    Connect to the SFTP endpoint which receives archives from CRIS and
+    download them into a temporary directory.
+
+    Returns path of temporary directory as a string
+    """
+
+    with SshKeyTempDir() as key_directory:
+        write_key_to_file(key_directory + "/id_ed25519", SFTP_ENDPOINT_SSH_PRIVATE_KEY + "\n") 
```

Comment:
You can see here I'm now using a class, defined below, to give me a way to get a directory that automatically is removed when it goes out of scope. A bit overkill now that this is in an ephemeral docker container, but I had the code all written, so it was easier to just do this fancy style.